src/apple/xcode-manager.ts

```diff
@@ -5,88 +5,144 @@
     (key) => {
+      const value = xcObjects.PBXNativeTarget?.[key];
       return (
         !key.endsWith('_comment') &&
-        xcObjects.PBXNativeTarget[key].name === targetName
+        typeof value !== 'string' &&
+        value?.name === targetName
       );
     },
   )[0];
-  const target = xcObjects.PBXNativeTarget[targetKey];
+  const target = xcObjects.PBXNativeTarget[targetKey] as
+    | PBXNativeTarget
+    | undefined;
 
-  xcObjects.XCConfigurationList[
-    target.buildConfigurationList
-  ].buildConfigurations.forEach((buildConfig: { value: string }) => {
-    const buildSettings =
-      xcObjects.XCBuildConfiguration[buildConfig.value].buildSettings;
+  if (!xcObjects.XCBuildConfiguration) {
+    xcObjects.XCBuildConfiguration = {};
+  }
+  if (!xcObjects.XCConfigurationList) {
+    xcObjects.XCConfigurationList = {};
+  }
+  const buildConfigurationListId = target?.buildConfigurationList ?? '';
+  const configurationList = xcObjects.XCConfigurationList?.[
+    buildConfigurationListId
+  ] as XCConfigurationList | undefined;
+  const buildListConfigurationIds =
+    configurationList?.buildConfigurations ?? [];
+  for (const buildListConfigId of buildListConfigurationIds) {
+    const config =
+      xcObjects.XCBuildConfiguration[buildListConfigId.value] ?? {};
+    if (typeof config === 'string') {
+      // Ignore comments
+      continue;
+    }
+
+    const buildSettings = config.buildSettings ?? {};
     buildSettings.DEBUG_INFORMATION_FORMAT = '"dwarf-with-dsym"';
     buildSettings.ENABLE_USER_SCRIPT_SANDBOXING = '"NO"';
-  });
+
+    config.buildSettings = buildSettings;
+    xcObjects.XCBuildConfiguration[buildListConfigId.value] = config;
+  }
 }
 
-function addSentrySPM(proj: any, targetName: string): void {
+function addSentrySPM(proj: Project, targetName: string): void {
   const xcObjects = proj.hash.project.objects;
 
-  const sentryFrameworkUUID = proj.generateUuid() as string;
-  const sentrySPMUUID = proj.generateUuid() as string;
+  const sentryFrameworkUUID = proj.generateUuid();
+  const sentrySPMUUID = proj.generateUuid();
 
-  //Check whether xcObjects already have sentry framework
+  // Check whether xcObjects already have sentry framework
   if (xcObjects.PBXFrameworksBuildPhase) {
     for (const key in xcObjects.PBXFrameworksBuildPhase || {}) {
-      if (!key.endsWith('_comment')) {
-        const frameworks = xcObjects.PBXFrameworksBuildPhase[key].files;
-        for (const framework of frameworks) {
-          if (framework.comment === 'Sentry in Frameworks') {
-            return;
-          }
+      const frameworkBuildPhase = xcObjects.PBXFrameworksBuildPhase[key];
+      if (key.endsWith('_comment') || typeof frameworkBuildPhase === 'string') {
+        // Ignore comments
+        continue;
+      }
+      for (const framework of frameworkBuildPhase.files ?? []) {
+        if (framework.comment === 'Sentry in Frameworks') {
```

Comment:
`m`: What is `Sentry in Frameworks`? Please add the info as a comment.

Reply:
It is set a couple of lines below as a comment and used to identify the framework in later runs.